gdpr/impl.go

```diff
@@ -109,120 +155,58 @@ func (p *permissionsImpl) allowSync(ctx context.Context, vendorID uint16, vendor
-	parsedConsent, vendor, err := p.parseVendor(ctx, vendorID, p.consent)
-	if err != nil {
-		return DenyAll, err
-	}
-
-	// vendor will be nil if not a valid TCF2 consent string
-	if vendor == nil {
-		if weakVendorEnforcement && parsedConsent.Version() == 2 {
-			vendor = vendorTrue{}
-		} else {
-			return DenyAll, nil
-		}
-	}
-
-	if !p.cfg.IsEnabled() {
-		return AllowBidRequestOnly, nil
-	}
+	purpose := consentconstants.Purpose(1)
+	enforcer := p.purposeEnforcerBuilder(purpose, bidder)
 
-	consentMeta, ok := parsedConsent.(tcf2.ConsentMetadata)
-	if !ok {
-		err = fmt.Errorf("Unable to access TCF2 parsed consent")
-		return DenyAll, err
-	}
-
-	permissions := AuctionPermissions{}
-	if p.cfg.FeatureOneEnforced() {
-		vendorException := p.cfg.FeatureOneVendorException(bidder)
-		permissions.PassGeo = vendorException || (consentMeta.SpecialFeatureOptIn(1) && (vendor.SpecialFeature(1) || weakVendorEnforcement))
-	} else {
-		permissions.PassGeo = true
-	}
-	if p.cfg.PurposeEnforced(consentconstants.Purpose(2)) {
-		enforceVendors := p.cfg.PurposeEnforcingVendors(consentconstants.Purpose(2))
-		vendorException := p.cfg.PurposeVendorException(consentconstants.Purpose(2), bidder)
-		permissions.AllowBidRequest = p.checkPurpose(consentMeta, vendor, vendorID, consentconstants.Purpose(2), enforceVendors, vendorException, weakVendorEnforcement)
-	} else {
-		permissions.AllowBidRequest = true
-	}
-	for i := 2; i <= 10; i++ {
-		enforceVendors := p.cfg.PurposeEnforcingVendors(consentconstants.Purpose(i))
-		vendorException := p.cfg.PurposeVendorException(consentconstants.Purpose(i), bidder)
-		if p.checkPurpose(consentMeta, vendor, vendorID, consentconstants.Purpose(i), enforceVendors, vendorException, weakVendorEnforcement) {
-			permissions.PassID = true
-			break
-		}
+	vendorInfo := VendorInfo{vendorID: vendorID, vendor: vendor}
+	if enforcer.LegalBasis(vendorInfo, bidder, consentMeta, Overrides{blockVendorExceptions: !vendorException}) {
+		return true, nil
 	}
-
-	return permissions, nil
+	return false, nil
 }
 
-const pubRestrictNotAllowed = 0
-const pubRestrictRequireConsent = 1
-const pubRestrictRequireLegitInterest = 2
-
-func (p *permissionsImpl) checkPurpose(consent tcf2.ConsentMetadata, vendor api.Vendor, vendorID uint16, purpose consentconstants.Purpose, enforceVendors, vendorException, weakVendorEnforcement bool) bool {
-	if consent.CheckPubRestriction(uint8(purpose), pubRestrictNotAllowed, vendorID) {
-		return false
-	}
+// allowBidRequest computes legal basis for a given bidder using the enforcement algorithms selected
+// by the purpose enforcer builder
+func (p *permissionsImpl) allowBidRequest(bidder openrtb_ext.BidderName, consentMeta tcf2.ConsentMetadata, vendorInfo VendorInfo) bool {
+	enforcer := p.purposeEnforcerBuilder(consentconstants.Purpose(2), bidder)
 
-	if vendorException {
+	if enforcer.LegalBasis(vendorInfo, bidder, consentMeta, Overrides{}) {
```

Comment:
Isn't this just the same as
`return enforcer.LegalBasis(vendorInfo, bidder, consentMeta, Overrides{})` ?